src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -10,12 +11,609 @@ namespace System.Buffers.Text.Tests
+            }
+        }
+
+        [Benchmark]
+        [InlineData("abcdef")] // standard parse
+        [InlineData("ffffffff")] // max value
+        [InlineData("0")] // min value
+        private static unsafe void ByteSpanToUInt32Hex(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            var utf8ByteSpan = new ReadOnlySpan<byte>(utf8ByteArray);
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed, 'X');
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        private static unsafe void ByteSpanToUInt32Hex_VariableLength()
+        {
+            int textLength = s_UInt32TextArrayHex.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_UInt32TextArrayHex[i]);
+            }
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed, 'X');
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        private static void ByteSpanToSByte_VariableLength()
+        {
+            int textLength = s_SByteTextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_SByteTextArray[i]);
+            }
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out sbyte value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData("0")]
```

Comment:
These test cases which are measuring StringToSByte should match the ones for ByteSpanToSByte tests so we can compare the results between string paring and `Span<byte>` parsing.

We should rename the tests to match what they do.

I see the following four tests for parsing each of the numeric types (Sbyte, Int32, Int64, etc.), and they should be named accordingly:
- StringToX
- SpanToX
- StringToX_VariableLength
- SpanToX_VariableLength

And then we have some with hex formatting.

This way, we can get the useful comparison of StringToX to SpanToX, for various test cases.